### PR TITLE
Another crossframe denoise fix

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

**Comment by jdent02** [comment]
#335

File: spool.py [hunk 1]
```diff
@@ -121,13 +121,20 @@ def spool_render(rman_version_short, rib_files, denoise_files, frame_begin, fram
                                     [('PixarRender', cmd_str)], 3)
             else:
                 denoise_options.append('-F 1 -L 1')
-                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - 3: frame_num]]
+                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - frame_begin - 2: frame_num - frame_begin + 1]]
                 write_cmd_task_line(f, 'Denoise frame %d' % (frame_num - 1),
                                     [('PixarRender', cmd_str)], 3)
             if frame_num == frame_end:
-                denoise_options.remove('-F 1 -L 1')
+                try:
```
Comment (bsavery) [inline spool.py]: Sorry why is there a try/catch necessary here?

Comment (bsavery) [inline spool.py]: Normally you'd do if x in list: list.remove(x)

**Comment by jdent02** [comment]
Main reason is a two frame animation failed to denoise properly.

**Comment by jdent02** [comment]
So I'll explain what this is doing: if there is a two frame animation the denoise options will have a '-L 1' option in there.  When it ran the old 'if frame==frame_end' loop it wasn't removing that, so you ended up with a denoise command that had two files and -L 1 and -F 1, so you'd get no output.

File: spool.py [hunk 2]
```diff
@@ -121,13 +121,13 @@ def spool_render(rman_version_short, rib_files, denoise_files, frame_begin, fram
                                     [('PixarRender', cmd_str)], 3)
             else:
                 denoise_options.append('-F 1 -L 1')
-                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - 3: frame_num]]
+                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - frame_begin - 2: frame_num - frame_begin + 1]]
                 write_cmd_task_line(f, 'Denoise frame %d' % (frame_num - 1),
                                     [('PixarRender', cmd_str)], 3)
             if frame_num == frame_end:
-                denoise_options.remove('-F 1 -L 1')
+                denoise_options = (x for x in denoise_options if x not in  ['-L 1', '-F 1 -L 1'])
```
Comment (bsavery) [inline spool.py]: did you want denoise_options to be a list or tuple?  List seems more appropriate

**Pull request closed without merging**